interview-questions.md

```diff
@@ -4,64 +4,77 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: a function is a block of code that executes any code found inside it's code block. You would use a function to have reusable code that you can call.
 
   Researched answer:
 
+  set of statements that performs a task or calculates a value. It allows you to define a block of code, give it a name and then execute it as many times as you want.
 
 
 2. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Filter is used to remove items from the array whereas map just iterates over it.
 
   Researched answer:
 
-
+  The map() method creates a new array with the results of calling a function for every array element. The filter() method creates an array filled with all array elements that pass a test implemented by the provided function
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: return is returning a value from a function. Console.log is printing what ever is inside the paranthesis into the console.
```

Comment:
yup! return is the final value of a function whereas console.log() is a developer's tool for debugging